chef/cookbooks/cpe_chrome/resources/cpe_chrome.rb

```diff
@@ -27,7 +27,8 @@ def manage_chrome
   end
 
   def manage_chrome_osx
-    prefs = node['cpe_chrome'].reject { |_k, v| v.nil? }
+    prefs = node['cpe_chrome']['profile'].reject \
+      { |_k, v| v.nil? || v.class == Chef::Node::ImmutableArray && v.empty? }
```

Comment:
I would raise caution on rejecting `ImmutableArray` since if we lay them down in a default attributes file that is the default class type. In `chef-shell`:
```
chef (12.18.31)> attributes_mode
chef:attributes (12.18.31)> default['thing'] = ['ohai!']
 => ["ohai!"]
chef:attributes (12.18.31)> exit
 => :attributes
chef (12.18.31)> node['thing'].class
 => Chef::Node::ImmutableArray
```
Just be careful that you're getting the behavior you want!

Some notes on style here:
* It's a bit more rubular to omit the `\` and drop the filter statement onto the next line after the `.`
* You can leverage `is_a?` to look at a class' type. Doing it this way will also save you a check for `nil?` since a `nil` value in ruby is also a `NilClass`. I.E.:
```
prefs = node['cpe_chrome']['profile'].
   reject { |_k, v| v.is_a?(Chef::Node::ImmutableArray) && v.empty? }
```

It's also not out of the ballpark to supply a single check for your filters and then chain them together, so another valid form of this would be:
```
prefs = node['cpe_chrome']['profile'].
   reject { |_k, v| v.is_a?(Chef::Node::ImmutableArray) }.
   reject { |_k, v| v.empty? }
```

Like mentioned below, if you wanted non-`nil` values you could dump all of them with `compact` before the `reject`s